python/ml4ir/base/data/ranklib_to_ml4ir.py

```diff
@@ -0,0 +1,60 @@
+import pandas as pd
+import argparse
+import logging
+
+#Constanst for the column names given to the query id and graded relevance.
+QUERY_ID_NAME = 'qid'
+RELEVANCE_NAME = 'relevance'
+
+
+def process_line(line, keep_additional_info):
+    if keep_additional_info == 1:
+        feature_values = line.replace('#', '').replace(' = ', ':').strip().split()
+    else:
+        feature_values = line.split('#')[0].strip().split()
+    feature_values[0] = RELEVANCE_NAME + ':' + feature_values[0]
+    r = {}
+    for fv in feature_values:
+        feat = fv.split(':')[0].strip()
+        val = fv.split(':')[1].strip()
+        if feat == RELEVANCE_NAME or feat == QUERY_ID_NAME:
+            r[feat] = val
+        else:
+            r['f_' + feat] = val
+    return r
+
+def convert(input_file, output_file, keep_additional_info, gl_2_clicks):
+    f = open(input_file, 'r')
+    rows = []
+
+    for line in f:
+        rows.append(process_line(line, keep_additional_info))
+    f.close()
+
+    df = pd.DataFrame(rows)
+    if gl_2_clicks == 1:
+        groups = df.groupby(QUERY_ID_NAME)
```

Comment:
@mohazahran Try using this method to convert to TFRecord. I think we should abstract the CSV data format here and jump to TFRecord as part of the `read()` method.

https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/data/tfrecord_writer.py#L80

Reply:
This is a good reference -> https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/data/csv_reader.py

Basically, write the dataframe to TFRecord and then read a TFRecordDataset

Reply:
Personally, I think converting to csv is more natural to me, plus it makes the script usable in more scenarios.
But, I can write another function that does that.

Reply:
Another function sounds good to me.